src/target/adi.c

```diff
@@ -380,7 +381,13 @@ bool adi_configure_mem_ap(adiv5_access_port_s *const ap)
 			 * valid debug components on AP0, so we have to have an exception
 			 * for this part family.
 			 */
-		if (ap->dp->target_designer_code != JEP106_MANUFACTURER_TEXAS || ap->base != 0xf0000002U) {
+		if (ap->dp->target_designer_code == JEP106_MANUFACTURER_TEXAS && ap->base == 0xf0000002U)
+			ignore_not_present = true;
+
+		if (ap->dp->target_designer_code == JEP106_MANUFACTURER_NORDIC && ap->base != 0x00000002U)
```

Comment:
This should probably be made an `else` to the above `if` (as in `else if`) - we see why you've reworked the logic in this way given the multiple different conditions to test for vendors having done different silly things here, but as this list could get quite long a cascading if-else check would allow the compiler a touch more efficiency in the long term.